### PR TITLE
Remove 'Clear profile' feature

### LINKED ISSUE #373: Remove 'Clear profile' feature
**Feature**: Remove the "Clear profile" functionality

**Problem**: Clear profile only clears fields without deleting the profile, causing confusion about its purpose

**Solution**: Remove the "Clear profile" button/feature entirely from the app

Part of #361

### PR DESCRIPTION
**Pull request by ziyad-elarroussi**
## Description
Remove the "Clear profile" button from the profile edit screen. This button **only cleared profile fields** (reset name, bio, location, hobbies to defaults) without actually deleting the profile or account, causing confusion about its purpose.

**Note:** This is NOT the account deletion feature in Settings (which permanently deletes everything). That functionality remains untouched.

**Related Issue:** Closes #373

---

## Changes

**ProfileScreen.kt:**
- Removed "Clear Profile" button from `EditProfileContent`
- Removed `DeleteProfileConfirmationDialog` component
- Removed delete confirmation dialog rendering logic

**ProfileViewModel.kt:**
- Removed `showDeleteConfirmation` property
- Removed `showDeleteDialog()` method
- Removed `hideDeleteDialog()` method
- Removed `deleteProfile()` method (54 lines - the "clear fields" logic)

**ProfileScreenTest.kt:**
- Removed 13 tests related to the "Clear Profile" button and dialog
- Removed mock setup for `showDeleteConfirmation` in remaining tests

---

## Testing
- [x] Unit tests pass
- [x] Instrumentation tests pass (528 tests)
- [x] Manual testing completed
- [x] Coverage maintained (≥80%)

**Test summary:** Removed 13 obsolete tests. All 528 instrumentation tests pass.

---

## Checklist
- [x] Sufficient documentation and minimal inline comments
- [x] All tests passing
- [x] No new warnings
- [x] If related issue exists: issue has all labels, fields, and milestone filled

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/src/main/java/com/swent/mapin/ui/profile/ProfileScreen.kt [hunk 1]
```diff
@@ -198,13 +197,6 @@ fun ProfileScreen(
                 },
                 onDismiss = { viewModel.toggleBannerSelector() })
           }
-
-          // Delete Confirmation Dialog
-          if (viewModel.showDeleteConfirmation) {
-            DeleteProfileConfirmationDialog(
-                onConfirm = { viewModel.deleteProfile() },
-                onDismiss = { viewModel.hideDeleteDialog() })
-          }
```
Comment (naderbahri) [inline app/src/main/java/com/swent/mapin/ui/profile/ProfileScreen.kt]: **Suggestion**: Either restore the dialog and update it to follow the app's current UX guidelines, or add a clear replacement with documentation in the PR describing where and how deletion is now performed (server-side endpoint, settings screen, or remote API). Add a migration note in the PR description and link any related issue.

File: app/src/androidTest/java/com/swent/mapin/ui/profile/ProfileScreenTest.kt [hunk 2]
```diff
@@ -1,10 +1,7 @@
 package com.swent.mapin.ui.profile
 
-import androidx.compose.foundation.rememberScrollState
-import androidx.compose.foundation.verticalScroll
```
Comment (naderbahri) [inline app/src/androidTest/java/com/swent/mapin/ui/profile/ProfileScreenTest.kt]: **Suggestion**: If the UI is still scrollable, keep tests that exercise scrolling and use rememberScrollState and verticalScroll in test harnesses. If the layout changed to non-scrollable, update tests accordingly and assert the new layout constraints explicitly.

File: app/src/androidTest/java/com/swent/mapin/ui/profile/ProfileScreenTest.kt [hunk 3]
```diff
@@ -656,71 +457,4 @@ class ProfileScreenTest {
...
-    var dismissed = false
-    composeTestRule.setContent {
-      MaterialTheme {
-        DeleteProfileConfirmationDialog(onConfirm = {}, onDismiss = { dismissed = true })
-      }
-    }
-
-    composeTestRule.onNodeWithText("Cancel").performClick()
-    assert(dismissed)
-  }
```
Comment (naderbahri) [inline app/src/androidTest/java/com/swent/mapin/ui/profile/ProfileScreenTest.kt]: **Suggestion**: If reintroducing the dialog, extract a smaller isolated composable and add lightweight unit tests verifying content and actions. Use string resource matching instead of exact English strings where possible to keep tests robust to localization.

File: app/src/androidTest/java/com/swent/mapin/ui/profile/ProfileScreenTest.kt
Comment (naderbahri) [inline app/src/androidTest/java/com/swent/mapin/ui/profile/ProfileScreenTest.kt]: **Error handling during profile deletion**

- **Issue**: No tests were left verifying what happens when deleteProfile fails due to network errors or permission issues. This is a common edge case for destructive operations.

- **Suggestion**: Add tests at the ViewModel layer to simulate deletion failure and confirm appropriate UI state and error messages are surfaced.

**Review by naderbahri — CHANGES_REQUESTED** [review]
This PR removes the clear-profile UI I implemented a long time ago and a large collection of related tests, which substantially reduces coverage for a destructive flow and introduces risk of regressions and dead code. The change is high impact and needs explicit justification, replacement tests for the new flow, and a clear migration note in the PR description.
Overall, it's a good PR and I'm ok with removing the clear-profile feature.

File: app/src/main/java/com/swent/mapin/ui/profile/ProfileScreen.kt [hunk 4]
```diff
@@ -30,7 +30,6 @@ import androidx.compose.foundation.verticalScroll
 import androidx.compose.material.icons.Icons
 import androidx.compose.material.icons.automirrored.filled.ArrowBack
 import androidx.compose.material.icons.filled.Check
-import androidx.compose.material.icons.filled.Delete
```
[the remark this answers is not in the document]
Comment (ziyad-elarroussi) [reply app/src/main/java/com/swent/mapin/ui/profile/ProfileScreen.kt]: Already handled by ktfmt auto-formatting. No action needed.

File: app/src/main/java/com/swent/mapin/ui/profile/ProfileScreen.kt (hunk 1 above)
Comment (ziyad-elarroussi) [reply app/src/main/java/com/swent/mapin/ui/profile/ProfileScreen.kt]: This wasn't actual server deletion, just a confusing  "clear fields" button. Ive checked manually and the profile deletion still shows the Confirmation Dialog.

File: app/src/androidTest/java/com/swent/mapin/ui/profile/ProfileScreenTest.kt [hunk 5]
```diff
@@ -165,26 +161,6 @@ class ProfileScreenTest {
...
-    every { mockViewModel.showDeleteConfirmation } returns true
-
-    composeTestRule.setContent {
-      MaterialTheme {
-        ProfileScreen(onNavigateBack = {}, onNavigateToSignIn = {}, viewModel = mockViewModel)
-      }
-    }
-
-    composeTestRule.onNodeWithText("Confirm Deletion").assertIsDisplayed()
-  }
```
[the remark this answers is not in the document]
Comment (ziyad-elarroussi) [reply app/src/androidTest/java/com/swent/mapin/ui/profile/ProfileScreenTest.kt]: No deletion flow exists or is planned. It's not a server-side flow.

File: app/src/main/java/com/swent/mapin/ui/profile/ProfileScreen.kt [hunk 6]
```diff
@@ -623,30 +615,6 @@ internal fun EditProfileContent(viewModel: ProfileViewModel) {
...
-                          imageVector = Icons.Default.Delete,
-                          contentDescription = "Clear Profile",
-                          modifier = Modifier.size(20.dp))
-                      Spacer(modifier = Modifier.width(8.dp))
-                      Text(
-                          "Clear Profile",
-                          style = MaterialTheme.typography.bodyLarge,
-                          fontWeight = FontWeight.Bold)
-                    }
-              }
```
[the remark this answers is not in the document]
Comment (ziyad-elarroussi) [reply app/src/main/java/com/swent/mapin/ui/profile/ProfileScreen.kt]: Done, and documented the changes in the PR as well.

File: app/src/androidTest/java/com/swent/mapin/ui/profile/ProfileScreenTest.kt
Comment (naderbahri) [inline app/src/androidTest/java/com/swent/mapin/ui/profile/ProfileScreenTest.kt]: Thanks for the changes

**Review by naderbahri — APPROVED** [review]
